src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToItemProvider.Searcher.cs

```diff
@@ -30,15 +32,24 @@ private class Searcher
                 ItemDisplayFactory displayFactory,
                 INavigateToCallback callback,
                 string searchPattern,
+                bool searchCurrentDocument,
                 CancellationToken cancellationToken)
             {
                 _solution = solution;
                 _displayFactory = displayFactory;
                 _callback = callback;
                 _searchPattern = searchPattern;
+                _searchCurrentDocument = searchCurrentDocument;
                 _cancellationToken = cancellationToken;
                 _progress = new ProgressTracker(callback.ReportProgress);
                 _asyncListener = asyncListener;
+
+                if (_searchCurrentDocument)
```

Comment:
I went back and forth myself.  In the end i preferred the explicitness of this appraoch.